c/auth.c

```diff
@@ -625,6 +615,35 @@ int validate_signature_solana(void *prefilled_data, const uint8_t *sig,
     return err;
 }
 
+#ifdef LIBECC_ONLY
+int validate_signature_secp256r1(void *prefilled_data, const uint8_t *sig,
```

Comment:
Move this to a new file: auth_libecc.c. The binary auth_libecc is built from this file.